spec/index.bs

```diff
@@ -6361,6 +6361,12 @@ and <dfn dfn>vertex data type</dfn> for the component.
   * `sint` = signed int
   * `float` = floating point
 
+The multi-component formats specify the number of components after "x".
+As such, {{GPUVertexFormat/"sint32x3"}} denotes a 3-component vector of `i32` values in the shader.
+
+See [[#vertex-processing]] for additional information about how vertex formats are exposed in the
+shader.
```

Comment:
```suggestion
shader, for example when their types and vector length don't match.
```

Hm, the vertex processing section doesn't seem to say anything about when the scalar type doesn't match? e.g. sint16x2-> f32
Maybe I missed it.
Or does something validate that? (I don't see anything.)

Reply:
Reached out to David about the type mismatch thing. So it seems the shader must always use the same base type as the vertex attribute (float, unsigned int, signed int) but it's an open question as to how much conversion between sizes of the same base type is allowed, especially regarding `f16`. As such I'm just going to omit that for now.

Reply:
I looked at Vulkan.   For pipeline IO:

* Right, It requires component-kind to match (what it calls "base type"):  unsigned to unsigned, signed to signed, floating to floating.
* Component counts are flexible, as you say.
* And, what's now clear to me:   lots of flexibility in the component type in the shader.  So, for example, a floating component can map to f32 or f16. 
         * mapping to 16bit input/output types requires 'storageInputOutput16' feature.   In WGSL f16 discussions we agreed to hid this detail: When the underlying Vulkan device doesn't support this we'll polyfill in the shader.


Reference: 22.4.1 "Vertex Input Extraction" https://renderdoc.org/vkspec_chunked/chap22.html#fxvertex-input-extraction 

> For attributes that are not 64-bit data types, each component is converted to the format of the input variable based on its type and size (as defined in the [Format Definition](https://renderdoc.org/vkspec_chunked/chap43.html#formats-definition) section for each [VkFormat](https://renderdoc.org/vkspec_chunked/chap43.html#VkFormat)), using the appropriate equations in [16-Bit Floating-Point Numbers](https://renderdoc.org/vkspec_chunked/chap3.html#fundamentals-fp16), [Unsigned 11-Bit Floating-Point Numbers](https://renderdoc.org/vkspec_chunked/chap3.html#fundamentals-fp11), [Unsigned 10-Bit Floating-Point Numbers](https://renderdoc.org/vkspec_chunked/chap3.html#fundamentals-fp10), [Fixed-Point Data Conversion](https://renderdoc.org/vkspec_chunked/chap3.html#fundamentals-fixedconv), and [Shared Exponent to RGB](https://renderdoc.org/vkspec_chunked/chap16.html#textures-sexp-RGB). Signed integer components smaller than 32 bits are sign-extended.

Reply:
(Also, 64-bit has less flexibility:  64bit vertex components must map to 64bit shader components, and vice versa.  But we don't have that problem in WebGPU yet)